src/web_peer_connection.cpp

```diff
@@ -86,16 +86,16 @@ web_peer_connection::web_peer_connection(peer_connection_args const& pack
 	shared_ptr<torrent> tor = pack.tor.lock();
 	TORRENT_ASSERT(tor);
 
-	// we always prefer downloading 1 MiB chunks
-	// from web seeds, or whole pieces if pieces
-	// are larger than a MiB
-	int preferred_size = 1024 * 1024;
+	// if the web server is known not to support keep-alive. request 4MiB
+	// but we want to have at least piece size to prevent block based requests
+	int const min_size = std::max((web.supports_keepalive ? 1 : 4) * 1024 * 1024,
```

Comment:
Actually something else is determining the size. If we pick the rarest piece for a BT peer we would get a gap on the next http request.

And I don't want to multiply max request bytes for non keepalive because if you have already a large size you will get a huge new size which would lock to much pieces of the file and speed for multi connection users will be reduced because picker has no pieces left to pick.

The point is, what is the different between 20s per request and 40s per request? It would not matter because one connection each 20s does already not hurt anybody.

But real aggressive user which selects something like 1MiB or less should be forced to use larger sizes because such user would give the bittorrent client a bad reputation.